### PR TITLE
Cluster controller: Add support for destroy

### PR DESCRIPTION
**Pull request by alexbrand**
Hooked up the destroy functionality on the controller. We can now destroy clusters using the DELETE request.

### CODE REVIEW

**Comment by emmetthitz** [comment]
LGTM

File: pkg/controller/cluster.go [hunk 1]
```diff
@@ -101,23 +118,31 @@ func (c *clusterController) transition(cluster store.Cluster) store.Cluster {
 		}
 		cluster.CurrentState = installing
 		return cluster
+	case installed:
+		if cluster.DesiredState == destroyed {
```
Comment (dkoshkin) [inline pkg/controller/cluster.go]: Does this not yet support, mutating clusters?

Comment (emmetthitz) [reply pkg/controller/cluster.go]: I think that's out of scope for this PR for now.

Comment (dkoshkin) [reply pkg/controller/cluster.go]: Gotcha wanted to confirm that.

**Comment by dkoshkin** [comment]
LGTM

File: pkg/controller/cluster.go (hunk 1 above)
Comment (alexbrand) [reply pkg/controller/cluster.go]: We'll have to figure out the mechanics for _modify_. I think the server might have to set current state to `planned`.

**Pull request merged**